sdk/client-core/core/src/main/java/io/clientcore/core/util/ServerSentEventUtils.java

```diff
@@ -56,24 +55,50 @@ public static boolean isTextEventStreamContentType(String contentType) {
      * <p>
      * The passed {@link InputStream} will be closed by this method.
      *
-     * @param httpClient The {@link HttpClient} to send the {@code httpRequest} through.
-     * @param httpRequest The {@link HttpRequest} to send.
      * @param inputStream The {@link InputStream} to read data from.
      * @param listener The {@link ServerSentEventListener event listener} attached to the {@link HttpRequest}.
+     * @return The non-null {@link ServerSentResult result} in case of an error.
      * @throws IOException If an error occurs while reading the data.
      */
-    public static void processTextEventStream(HttpClient httpClient, HttpRequest httpRequest,
-        InputStream inputStream, ServerSentEventListener listener) throws IOException {
+    public static ServerSentResult processTextEventStream(InputStream inputStream,
+        ServerSentEventListener listener) throws IOException {
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
-            RetrySSEResult retrySSEResult = processBuffer(reader, listener);
+            return processBuffer(reader, listener);
+        }
+    }
+
+    /**
+     * Retries reconnect to the {@link HttpRequest} if the {@link ServerSentResult#getRetryAfter()}
+     * and {@link ServerSentResult#getLastEventId()} are not null.
+     * <p>
+     * The method will retry the request with the last event id set and after the retry time.
+     * </p>
+     * @param serverSentResult The {@link ServerSentResult result} of the retry.
+     * @param httpRequest The {@link HttpRequest} to send.
+     * @return {@code true} if the request was retried; {@code false} otherwise.
+     */
+    public static boolean retryReconnect(ServerSentResult serverSentResult, HttpRequest httpRequest) {
+        if (!shouldDefaultRetry(serverSentResult.getRetryAfter(), serverSentResult.getLastEventId())) {
+            // Retry the request with the last event id set
+            httpRequest.getHeaders()
+                .add(LAST_EVENT_ID, serverSentResult.getLastEventId());
 
-            if (retrySSEResult != null && !retryExceptionForSSE(retrySSEResult, listener, httpRequest)
-                && !Thread.currentThread().isInterrupted()) {
-                httpClient.send(httpRequest);
+            // Retry the request after the retry time
+            try {
+                Thread.sleep(serverSentResult.getRetryAfter().toMillis());
+            } catch (InterruptedException ignored) {
+                // ignored
             }
+            return true;
+        } else {
+            return false;
         }
     }
 
+    private static boolean shouldDefaultRetry(Duration retryAfter, String lastEventId) {
+        return retryAfter == null || lastEventId == null;
```

Comment:
Is a missing last event ID really an invalid retry state? Wouldn't that just be a retry without a starting point specifier?

Reply:
Yeah we could update to just check for availability of retry After, wanted to put this out for default and take opinions.